File: 백준/Bronze/4153. 직각삼각형/직각삼각형.cc
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
    int a,b,c;

    while(1){
        cin >> a >> b >> c;
        if(a==0 && b==0 && c==0) break;

        int longest=max({a,b,c});
        int shortest=min({a,b,c});
        int rest;

        if(a!=longest&&a!=shortest) rest=a;
        if(b!=longest&&b!=shortest) rest=b;
        if(c!=longest&&c!=shortest) rest=c;

        if((shortest*shortest)+(rest*rest)==(longest*longest)){
            cout << "right" << "\n";
        }else{
            cout << "wrong" << "\n";
        }

    }

    return 0;
}

File: 백준/Bronze/4153. 직각삼각형/직각삼각형.js
const fs = require("fs");
const path = require("path");

const input = fs
  .readFileSync(
    process.platform === "linux"
      ? "/dev/stdin"
      : path.join(__dirname, "/input.txt")
  )
  .toString()
  .trim()
  .split("\n");

const solution = (input) => {
  const answer = input.map((v) => {
    const arr = v.split(" ").map(Number);
    const c = Math.max(...arr);
    const [a, b] = arr.filter((v) => v !== c);
    if (Math.pow(a, 2) + Math.pow(b, 2) === Math.pow(c, 2)) return "right";
    else return "wrong";
  });
  console.log(answer.join("\n"));
};

solution(input.slice(0, -1));
